chore: add documentation for adding a component library to the creator (#4883)
microsoft/fast: sites/fast-creator/specs/site.spec.md

Each library will be defined by a configuration which includes the web components themselves, data examples that will be used when a component is added, the tag name of the components and the display name. See the [typings](./app/configs/typings.ts) file for the interface.

> When a user adds a library to a project, this library configuration will be dynamically imported. Available libraries will be referenced in a dictionary file, this file will be used to create a UI.

Reviewer: Maybe it's just I'm missing some context from prior doc sections, but I'm getting a little lost on what "this" is ("this library configuration", "this file"). Should that get re-worded?

Author: I've clarified by replacing "this" with what it refers to.

> When a user adds a library to a project, this library configuration will be dynamically imported. Available libraries will be referenced in a dictionary file, this file will be used to create a UI.

Author: ```suggestion
When a user adds a library to a project, the library configuration will be dynamically imported. Available libraries will be referenced in a dictionary file, the configuration file will be used to create a UI.
```

> When a user adds a library to a project, this library configuration will be dynamically imported. Available libraries will be referenced in a dictionary file, this file will be used to create a UI.

Reviewer: ```suggestion
When a user adds a library to a project, the library configuration will be dynamically imported. Available libraries will be referenced in a dictionary file, which will be used to create a UI.
```

Took a stab at what I think you mean that reads a bit more clearly to me. If it's not accurate please disregard

Author: It's accurate, I just pushed a change that updates it to something similar.

> When a user adds a library to a project, the library configuration will be dynamically imported. Available libraries will be referenced in a dictionary file, the configuration file will be used to create a UI.

Author: ```suggestion
When a user adds a library to a project, the library will be dynamically imported. Available libraries will be referenced in a dictionary file, and the library configuration file will be used to create a UI.
```
